vowpalwabbit/cb_algs.cc

```diff
@@ -147,9 +147,8 @@ base_learner* cb_algs_setup(options_i& options, vw& all)
     return nullptr;
 
   // Ensure serialization of this option in all cases.
-  if (!options.was_supplied("cb_type"))
+  if (options.ensure_default_dependency("cb_type", type_string))
```

Comment:
I find the prefix `ensure` on the method name a bit semantically overloaded. Combined with the method returning `true` or `false` depending on whether the key existed or not is a bit confusing. My first thought was that if `ensure_default_dependency` returns false then we failed ensure that key is set. Maybe just `set_default_if_missing` or something along those lines?

Reply:
The name is clunky so I wouldn't mind changing it. The boolean was mostly tacked on to be able to keep almost the same structure on the calling side